homeassistant/components/rako/light.py

```
@@ -0,0 +1,212 @@
+        return False
+
+    @property
+    def supported_features(self) -> int:
+        """Flag supported features."""
+        return SUPPORT_BRIGHTNESS
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn off the light."""
+        await self.async_turn_on(brightness=0)
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return device information about this Rako Light."""
+        return {
+            "identifiers": {(DOMAIN, self.unique_id)},
+            "name": self.name,
+            "manufacturer": "Rako",
+            "suggested_area": self._light.room_title,
+            "via_device": (DOMAIN, self.bridge.mac),
+        }
+
+
+class RakoRoomLight(RakoLight):
+    """Representation of a Rako Room Light."""
+
+    def __init__(self, bridge: RakoBridge, light: python_rako.RoomLight) -> None:
+        """Initialize a RakoLight."""
+        super().__init__(bridge, light)
+        self._light: python_rako.RoomLight = light
+
+    def _init_get_brightness_from_cache(self) -> int:
+        scene_of_room = self.bridge.scene_cache.get(self._light.room_id, 0)
+        brightness: int = convert_to_brightness(scene_of_room)
+        return brightness
+
+    @property
+    def name(self) -> str:
+        """Return the display name of this light."""
+        room_title: str = self._light.room_title
+        return room_title
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn on the light."""
+        brightness = kwargs.get(ATTR_BRIGHTNESS, 255)
+
+        try:
+            scene = convert_to_scene(brightness)
+            await asyncio.wait_for(
+                self.bridge.set_room_scene(self._light.room_id, scene), timeout=3.0
+            )
+
+        except (RakoBridgeError, asyncio.TimeoutError):
+            if self._available:
+                _LOGGER.error("An error occurred while updating the Rako Light")
+            self._available = False
+            return
```

I know we have requirements around logging once for higher integration quality levels, but we've also introduced a new rule about logging exceptions `HomeAssistantError`. I'd say re-throw as HomeAssistantError here.

---

The reason to do this is because automations should fail rather than silently succeed, e.g.